Added header support

arrygoo opened the pull request:
UI
- Form fields for adding new headers
- Form fields for existing headers
- event handler for change name (updates ApiConfig record)
- event handler for change value (updates ReactiveDict)
- Add headers to generated sample code

Publication
- Change format of parameter to take a single “options” param that looks like: {variables {"keyword":"TODO"}, headers: {...}}
- Error if passing in header name that’s not defined. Or silently ignore?
- HTTP.get() uses headers

Add example in seeds.js that uses headers??

Commit:
Added header support

arrygoo commented on dashboard/client/dashboard.js:
```diff
@@ -99,19 +112,53 @@ Template.dashboard.events({
+    ApiConfigs.update(activeConfigId.get(), {$pull: {headers: event.target.getAttribute('data-label')}} );
+  },
+  'keyup .headersName': function (event) {
+    // TODO update header in DB and in headerValus:
```
Not done.

pauldowman replied:
I see the issue. You want to set "headers.${n}" where n is the index, so something like `ApiConfigs.update(activeConfigId.get(). {$set: { "headers.0": event.target.value }})` if the particular header is the first header. But we don't actually know the index. You could maybe iterate through activeConfig().headers until you match. 

Actually I think it might be better to add the index onto the field using a data-\* attribute, then use it in here. I'll push a commit with that in a minute...

pauldowman replied:
OK I can't do that without merging this. But add `data-index='{{@index}}'` on the field in the template, and then here's the event handler:

``` javascript
  'keyup .headersName': function (event) {
    var index = event.target.dataset.index;
    var setter = {};
    setter[`headers.${index}`] = event.target.value;
    ApiConfigs.update(activeConfigId.get(), {$set: setter});
  },
```

arrygoo replied:
Amazing! Wish I had thought of this idea!

pauldowman commented on packages/npm-packages/.npm/package/npm-shrinkwrap.json:
```diff
@@ -80,6 +80,9 @@
       "version": "0.1.2"
+    },
+    "underscore": {
+      "version": "1.8.3"
```
Actually underscore should be available already. It's a standard Meteor package, and if it wasn't available already it can be added with `meteor add underscore`.

Commit:
removed underscore

Commit:
Fixed bug where other configs headers were being used in one config.

Commit:
• Added Paul's pull request suggestion
• Changed keyUp event to "change" to avoid interruption due to saving new header name

Commit:
Bug fix for configurations with no headers. (From pairing with Richard)
